src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/PrintfStressMessageFormatter.cs

```diff
@@ -0,0 +1,332 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Numerics;
+using System.Text;
+using Microsoft.Diagnostics.DataContractReader;
+using Microsoft.Diagnostics.DataContractReader.Contracts;
+
+namespace Microsoft.Diagnostics.DataContractReader;
+
+public sealed class PrintfStressMessageFormatter
```

Comment:
I'm not sure this really belongs in the Contracts assembly. It looks like a helper for use on top of the contracts, which seems more like part of a contract consumer - so the StressLogAnalyzer?

Reply:
My plan was to reuse this in other places that need to also implement stress message formatting, like an eventual implementation of SOS's DumpLog. It would need the exact same implementation.

Reply:
Okay. I still think it is a bit separate from the Contracts assembly, but I can see this as a reasonable place for now to avoid an entirely separate assembly and package to allow others to consume just for the one class. I think it is worth revisiting if we end up with a bunch more helpers worth reusing on top of the actual contracts.

Reply:
I'm fine revisiting this later and possibly adding another assembly or something.